packages/react-router-dev/vite/plugin.ts

```diff
@@ -2023,11 +2023,13 @@ function groupRoutesByParentId(manifest: ServerBuild["routes"]) {
   let routes: Record<string, Omit<ServerRoute, "children">[]> = {};
 
   Object.values(manifest).forEach((route) => {
-    let parentId = route.parentId || "";
-    if (!routes[parentId]) {
-      routes[parentId] = [];
+    if (route) {
+      let parentId = route.parentId || "";
+      if (!routes[parentId]) {
+        routes[parentId] = [];
+      }
+      routes[parentId].push(route);
     }
-    routes[parentId].push(route);
```

Comment:
* Remix `RouteManifest` was `Record<string, Route>`
* React Router's internal `RouteManifest` was more accurately `Record<string, Route | undefined>`

Switching to the stricter type means we have to introduce some defensiveness in existing Remix code - which feels more appropriate given the new fog of war behavior